benchmark/SerializerBenchmark/Serializers/OdinSerializer.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) All contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+extern alias odin;
+
+using Benchmark.Serializers;
+using odin::OdinSerializer;
+using odin::OdinSerializer.Utilities;
+
+#pragma warning disable SA1649 // File name should match first type name
+
+public class Odin_ : SerializerBase
+{
+    public override object Serialize<T>(T input)
+    {
+        using (var ctx = Cache<SerializationContext>.Claim())
+        {
```

Comment:
Maybe the bad perf that @TorVestergaard points out is because you're claiming this context every time. If a real app using Odin would do L16 just once and serialize many times, then we shouldn't include this step in every serialization under test.

Reply:
I'll try it out and see what difference it makes.

Reply:
Well, doing as you describe like this...

```csharp
// Copyright (c) All contributors. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

using Benchmark.Serializers;
using OdinSerializer;

#pragma warning disable SA1649 // File name should match first type name

public class Odin_ : SerializerBase
{
    private static readonly SerializationContext SerializationContext;
    private static readonly DeserializationContext DeserializationContext;

    static Odin_()
    {
        SerializationContext = new SerializationContext();
        SerializationContext.Config.SerializationPolicy = SerializationPolicies.Everything;

        DeserializationContext = new DeserializationContext();
        DeserializationContext.Config.SerializationPolicy = SerializationPolicies.Everything;
    }

    public override object Serialize<T>(T input)
    {
        return SerializationUtility.SerializeValue(input, DataFormat.Binary, SerializationContext);
    }

    public override T Deserialize<T>(object input)
    {
        return SerializationUtility.DeserializeValue<T>((byte[])input, DataFormat.Binary, DeserializationContext);
    }
}
```

...produces an impossibly small data size of five bytes.

Reply:
...no, doing it this way produces the same result.

```csharp
public class Odin_ : SerializerBase
{
    private readonly SerializationContext serializationContext;
    private readonly DeserializationContext deserializationContext;
    private readonly SerializationConfig serializationConfig;

    public Odin_()
    {
        serializationConfig = new SerializationConfig();
        serializationConfig.SerializationPolicy = SerializationPolicies.Everything;

        serializationContext = new SerializationContext();
        serializationContext.Config = serializationConfig;

        deserializationContext = new DeserializationContext();
        deserializationContext.Config = serializationConfig;
    }

    public override object Serialize<T>(T input)
    {
        return SerializationUtility.SerializeValue(input, DataFormat.Binary, serializationContext);
    }

    public override T Deserialize<T>(object input)
    {
        return SerializationUtility.DeserializeValue<T>((byte[])input, DataFormat.Binary, deserializationContext);
    }
}

```

That result, incidentally, is:

|            Method |         Serializer |      Mean | Error |  DataSize |   Gen 0 | Gen 1 | Gen 2 | Allocated |
|------------------ |------------------- |----------:|------:|----------:|--------:|------:|------:|----------:|
| AnswerDeserialize |              Odin_ | 51,332.0 ns |    NA |         - |  3.9673 |     - |     - |   12504 B |
|   AnswerSerialize |              Odin_ |    270.8 ns |    NA |    5.00 B |  0.0100 |     - |     - |      32 B |

Reply:
Thanks for trying. Any other suggestions, @TorVestergaard ?